backend/yagna/src/lib.rs

```diff
@@ -16,21 +18,45 @@ pub trait YagnaEngine: Engine {
         node_name: &str,
         wasm_url: &str,
         timeout: std::time::Duration,
+        subnet: Option<&String>,
     ) -> anyhow::Result<Demand>;
 }
 
 #[derive(Debug, Clone)]
 pub struct YagnaBackend {
     url: Option<String>,
     token: Option<String>,
+    subnet: Option<String>,
 }
 
 impl YagnaBackend {
     pub fn parse_url(url: &str) -> anyhow::Result<Option<Self>> {
+        if let Ok(url) = Url::parse(url) {
+            let scheme = url.scheme();
+            if scheme != "yagna" && scheme != "lwg" {
+                return Ok(None);
+            }
+            let mut token = None;
+            let mut subnet = None;
+            for (param, value) in url.query_pairs() {
+                match param.as_ref() {
+                    "token" | "appkey" => token = Some(value.into()),
```

Comment:
I'd reccomend single param for single purpose. No need to introduce ambiguity?
```suggestion
                    "appkey" => token = Some(value.into()),
```